TGAC-2014-genome-annotation
===========================

Course materials for the genome annotation part of the [TGAC SeqAhead course 2014](http://www.tgac.ac.uk/seqahead-challenges-for-plant-researchers/). Although these materials were written for the course, they should be useful to anyone wanting to learn about plant genome annotation.

By: Richard Smith-Unna ([Blahah](Https://github.com/Blahah)) <rds45@cam.ac.uk>.

### Using these materials

To download these materials:

```
git clone https://github.com/Blahah/TGAC-2014-genome-annotation.git
```

Or just view them online on Github: https://github.com/Blahah/TGAC-2014-genome-annotation

### Scope

The course covers both structural and functional annotation of plant genomes. Much of the material is applicable to eukaryotic genome annotation in general, but some content is specific to plants.

### Prerequisites

To get the most out of this course you should:

- be familiar with genomics and bioinformatics in general (what a genome is, assembly, RNAseq, etc.)
- be comfortable in Linux
- be able to run commands in the terminal
- know how to install software

### Outcomes

By the end of the course you should:

- understand what structural and functional annotation are
- be aware of the main theoretical approaches to these tasks
- know which software to use
- understand what outputs to generate
- be able to perform a first-draft annotation of a newly assembled genome

### Resources

- Slides \[[source](talk)|[view online]()\]
- [Genome annotation glossary](https://github.com/Blahah/TGAC-2014-genome-annotation/wiki/Genome-annotation-glossary)
- [Reading list](https://github.com/Blahah/TGAC-2014-genome-annotation/wiki/Reading-list)
- [Software list](https://github.com/Blahah/TGAC-2014-genome-annotation/wiki/Software-list)
- [Exercises](exercises)
- [Genome annotation walkthrough](https://github.com/Blahah/TGAC-2014-genome-annotation/wiki/Annotation-pipeline)
