<!doctype html>
<html lang="en">

<head>
	<meta charset="utf-8">

	<title>Annotating Plant Genomes</title>

	<meta name="description" content="A framework for easily creating beautiful presentations using HTML">
	<meta name="author" content="Richard Smith-Unna">

	<meta name="apple-mobile-web-app-capable" content="yes" />
	<meta name="apple-mobile-web-app-status-bar-style" content="black-translucent" />

	<meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no">

	<link rel="stylesheet" href="css/reveal.min.css">
	<link rel="stylesheet" href="css/theme/simple.css" id="theme">

	<!-- For syntax highlighting -->
	<link rel="stylesheet" href="lib/css/zenburn.css">

	<!-- If the query includes 'print-pdf', include the PDF print sheet -->
	<script>
		if( window.location.search.match( /print-pdf/gi ) ) {
			var link = document.createElement( 'link' );
			link.rel = 'stylesheet';
			link.type = 'text/css';
			link.href = 'css/print/pdf.css';
			document.getElementsByTagName( 'head' )[0].appendChild( link );
		}
	</script>

		<!--[if lt IE 9]>
		<script src="lib/js/html5shiv.js"></script>
		<![endif]-->
	</head>

	<body>

		<div class="reveal">

			<div class="slides">
				<section class="" data-id="61a12618a04b9f399262b26afa67b237">
					<h1>
						<span style="text-transform: uppercase;">Annotating<br>
						</span>
						<span>PLANT</span><br>
						<span>genomes</span>
					</h1>
					<p style="text-align: center;">
						<span>Richard Smith-Unna&nbsp;</span>
					</p>
					<p style="text-align: center;">
						<img src="https://raw.githubusercontent.com/HibberdLab/hibberdlab.github.com/master/assets/logo.png" style="width: 238.08599414658983px; height: 53.17285855227594px; max-height: none; max-width: none;">
						<img src="https://s3.amazonaws.com/media-p.slid.es/uploads/richardsmith-unna/images/406632/Screenshot_2014-05-15_13.39.02.png">
					</p>
				</section>
				<section class="" data-id="37f4442ad250b11d4e5474f37fdc08b8">
					<h2>What we'll cover</h2>
					<div>
						<ul>
							<li>what annotation is</li>
							<li>what the important steps are</li>
							<li>how those steps work</li>
							<li>what software to use</li>
							<li>what to do with the data</li>
						</ul>
						<div style="text-align: left;">
							<br>
						</div>
					</div>
					<div style="text-align: center;">and we'll get our hands dirty</div>
					<div style="text-align: center;">with some real (rice) data</div>
				</section>
				<section class="" data-id="e8145682a5988e36a8b03f23643e6ac3">
					<h2>RESOURCES</h2>
					<div>
						<ul>
							<li>these slides</li>
							<li>glossary</li>
							<li>reading list</li>
							<li>software list</li>
							<li>exercises</li>
							<li>step-by-step walkthrough</li>
						</ul>
						<div style="text-align: left;">
							<br>
						</div>
					</div>
					<div style="text-align: center;">all online at:</div>
					<div style="text-align: center;">
						<a href="https://github.com/Blahah/TGAC-2014-genome-annotation">https://github.com/Blahah/TGAC-2014-genome-annotation</a>
					</div>
				</section>
				<section class="" data-id="2e9e2ed6e395d883d474fe4e2f908d18">
					<h1>
						<font style="font-size: 72px;">structural annotation</font>
					</h1>
					<div>
						<ul>
							<li>
								<span style="text-align: center;">where are the genes?</span>
							</li>
							<li>
								<span style="text-align: center;">what features do they have, and where?</span>
							</li>
							<li>
								<span style="text-align: center;">what products do they encode?</span>
							</li>
						</ul>
					</div>
				</section>
				<section class="" data-id="922f79dfce6cf9830082853a35a12cef">
					<h1>
						<font style="font-size: 72px;">FUNCTIONAL ANNOTATION</font>
					</h1>
					<div>
						<div style="font-style: normal; font-variant: normal;">
							<ul>
								<li>
									<span style="text-align: center;">what is the biological role of those gene products?</span>
									<br>
								</li>
								<li style="font-size: 36.457374572753906px;">
									<span style="text-align: center;">what should we call them?</span>
								</li>
								<li style="font-size: 36.457374572753906px;">
									<span style="text-align: center;">how is the gene regulated?</span>
								</li>
							</ul>
						</div>
					</div>
				</section>
				<section class="" data-id="3078cc33c5cfdc065c363666bdc10399">
					<h2>don't expect too much</h2>
					<div>
						<br>
					</div>
					<div>Annotation is an imperfect science.</div>
					<div>
						<br>
					</div>
					<div>Even the best model genomes (e.g. human)</div>
					<div>have&nbsp;<span>very incomplete annotations:</span>
					</div>
					<div>
						<ul>
							<li>over 90% of protein loci correctly identified</li>
							<li>only ~50% have structure correctly annotated</li>
							<li>~30% lacking any functional annotation</li>
						</ul>
					</div>
				</section>
				<section class="" data-id="6e3c1a5f70bcc60ed7832cdd62c3d0d5">
					<h2>
						<br>
					</h2>
					<h2>EXAMPLE workflow</h2>
					<div>
						<br>
					</div>
					<img src="https://s3.amazonaws.com/media-p.slid.es/uploads/richardsmith-unna/images/395365/genome_annotation_process.png" style="width: 598.2420519724408px; height: 345.14233110544103px; max-height: none; max-width: none;">
				</section>
				<section class="" data-id="24dd3612ca23ab9aa28eb16913e45847">
					<h2>REPEATS</h2>
					<div>
						<br>
					</div>
					<div>refers to multiple different biological phenomena</div>
					<div>
						<br>
					</div>
					<div>
						<ul>
							<li>low complexity sequences</li>
							<ul>
								<li>
									<font style="font-size: 24px;">AAAAATTTTTTAAAAAAAAACCAAAA</font>
								</li>
							</ul>
							<li>short tandem repeats</li>
							<ul>
								<li>
									<font style="font-size: 24px;">ACACACACACACACAACACACACAC</font>
								</li>
							</ul>
							<li>transposable elements&nbsp;</li>
							<ul>
								<li>LINEs, SINEs and viruses</li>
								<li>80% of maize &amp; wheat genomes!</li>
							</ul>
						</ul>
					</div>
				</section>
				<section class="" data-id="dd4254cfc825e47e342fe4ba5c79d935">
					<h2>REPEATS ARE IMPORTANT</h2>
					<h3>
						<span style="font-size: 55.8px;">
							<br>
						</span>
					</h3>
					<h3>
						<span style="font-size: 55.8px;">GOOD</span>
						<br>
					</h3>
					<div>
						<div style="font-size: 36.185611724853516px; font-style: normal; font-variant: normal;">
							<ul>
								<li>useful in phylogenetics and forensics</li>
								<li>fascinating to study in their own right</li>
								<li>cause genome expansions</li>
								<li>affect evolution of genes and regulatory elements</li>
							</ul>
						</div>
					</div>
				</section>
				<section class="" data-id="f4098bf992ef54c369386c71c5e59464">
					<h2>REPEATS ARE IMPORTANT</h2>
					<div>
						<br>
					</div>
					<h3>bad</h3>
					<div>
						<ul>
							<li>
								<span style="text-align: center;">sometimes look like real genes</span>
								<br>
							</li>
							<li>
								<span style="text-align: center;">cause spurious alignments</span>
								<br>
							</li>
							<li>
								<span style="text-align: center;">want to prevent them affecting analysis</span>
							</li>
						</ul>
					</div>
				</section>
				<section class="" data-id="6be10bd00d8a3afb6df487a1c975d211">
					<h2>
						<span style="font-size: 75.96px;">IDENTIFYING REPEATS</span>
						<br>
					</h2>
					<div>
						<br>
					</div>
					<h3>homology</h3>
					<div>do they look like known repetitive sequences?</div>
					<div>
						<br>
					</div>
					<h3>DE-NOVO</h3>
					<div>compare a genome to itself and cluster</div>
					<div>OR</div>
					<div>use regular expressions</div>
				</section>
				<section class="" data-id="470b398fead9112e02cd316bbf08014e">
					<h2>
						<span>MASKING</span>&nbsp;REPEATS</h2>
						<div>
							<br>
						</div>
						<div>SOFT:&nbsp;<span>ACAGTAAaaaattttaaaaACTGCATGCAAAT</span>
						</div>
						<div>
							<br>
						</div>
						<div>HARD:&nbsp;<span>ACAGTAANNNNNNNNACTGCATGCAAAT</span>
						</div>
						<div>
							<span>
								<br>
							</span>
						</div>
						<div>
							<span>
								<font style="font-size: 24px;">exercise 1</font>
							</span>
						</div>
					</section>
					<section class="" data-id="86f3d7e26314663901270282fd722cd9">
						<h2>GENES</h2>
						<div>
							<span>shared structure allows de-novo prediction&nbsp;</span>
							<br>
						</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>start and stop codons</li>
								<li>splice-sites</li>
								<li>promoters</li>
							</ul>
							<div style="text-align: left;">
								<br>
							</div>
						</div>
						<div style="text-align: center;">but it's not very accurate (~40-50%)</div>
					</section>
					<section class="" data-id="79edce6b6f66080b83ef6d9302532498">
						<h2>GENES</h2>
						<div>
							<br>
						</div>
						<div>using data from biological experiments</div>
						<div>gives evidence-based predictions</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>align proteins from related species</li>
								<li>ESTs, cDNAs from same or very close spp.</li>
								<li>assembled transcripts</li>
								<li>RNAseq reads</li>
							</ul>
							<div style="text-align: center;">
								<br>
							</div>
						</div>
						<div style="text-align: center;">much more accurate (~90%)</div>
						<div style="text-align: center;">
							<br>
						</div>
					</section>
					<section class="" data-id="1b2abba64242e7224d121e17113079f6">
						<h2>WHAT DOES THE EVIDENCE SAY?</h2>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>spliced protein alignment:</li>
								<ul>
									<li>evidence for exons</li>
								</ul>
								<li>spliced transcript alignment:</li>
								<ul>
									<li>more evidence for exons</li>
									<li>alternative transcripts</li>
									<li>&nbsp;shows gene is expressed</li>
									<li>compare with protein for UTR evidence</li>
								</ul>
								<li>RNAseq reads:</li>
								<ul>
									<li>alternative transcripts</li>
									<li>expression levels and patterns</li>
								</ul>
							</ul>
						</div>
					</section>
					<section class="" data-id="2d54e319f0bf2e84b1042f472e9b727f">
						<h2>STRUCtURAL aNNOTATION PIPELINES</h2>
						<div>
							<ol>
								<li>
									<span style="text-align: center;">take in multiple sources of evidence</span>
									<br>
								</li>
								<li>
									<span style="text-align: center;">balance them based on confidence</span>
									<br>
								</li>
								<li>
									<span style="text-align: center;">create a merged annotation</span>
									<br>
								</li>
							</ol>
							<div>
								<br>
							</div>
						</div>
						<div>examples:</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>MAKER (runs all the tools for you!)</li>
								<li>AUGUSTUS</li>
								<li>PASA-EVM</li>
							</ul>
						</div>
					</section>
					<section class="" data-id="497491d95c1ae1673988eda30f1b2d43">
						<h2>THE EVIDENCE TRAIL</h2>
						<div>however you annotate,</div>
						<div>you need to record the evidence</div>
						<div>
							<br>
						</div>
						<div>pipelines do this for you</div>
						<div>
							<br>
						</div>
						<div>present the evidence in a genome browser</div>
						<div>
							<br>
						</div>
						<div>
							<span style="font-size: 24.47850227355957px; font-style: normal; font-variant: normal;">exercise 2</span>
							<br>
						</div>
					</section>
					<section class="" data-id="44bd4a2f618be8460a238daceef9bf0f">
						<h2>FUNCTIONAL ANNOTATION</h2>
						<div>
							<span style="text-align: left;">homology (sequence alignment)</span>
							<br>
						</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>use predicted protein sequences</li>
								<li>perform pairwise alignment against related spp.</li>
								<li>transfer functional info between species</li>
								<li>e.g. CRB-BLAST</li>
							</ul>
						</div>
					</section>
					<section class="" data-id="cf680a2e5e954f72b933e90fea4fae8b">
						<h2>FUNCTIONAL ANNOTATION</h2>
						<div>
							<br>
						</div>
						<div>homology (sequence profiles)</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>build a probabilistic model of the sequence</li>
								<li>search the model into a database of models</li>
								<li>can be whole-sequence or domain specific</li>
								<li>e.g. HMMER</li>
							</ul>
						</div>
					</section>
					<section class="" data-id="67bbb59e666647f4160d736858bb04c0">
						<h2>FUNCTIONAL ANNOTATION PIPELINES</h2>
						<div>
							<ol>
								<li>
									<span style="text-align: center;">use multiple methods</span>
									<br>
								</li>
								<li>
									<span style="text-align: center;">search multiple databases and references</span>
									<br>
								</li>
								<li>
									<span style="text-align: center;">collect the evidence</span>
									<br>
								</li>
							</ol>
							<div>
								<br>
							</div>
						</div>
						<div>example tools:</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>InterProScan</li>
								<li>Mercator</li>
								<li>Trinotate</li>
							</ul>
							<div style="text-align: left;">
								<br>
							</div>
						</div>
						<p style="text-align: center;">
							<font style="font-size: 24px;">
								<span>exercise 3</span>
							</font>
						</p>
					</section>
					<section class="" data-id="164f0562d530b8bd49bd9769a78ce7de">
						<h2>MANUaL CURATION</h2>
						<div>incorporate structural information</div>
						<div>
							<br>
						</div>
						<div>important to keep the evidence trail</div>
						<div>
							<br>
						</div>
						<div>gene family experts required for high confidence</div>
						<div>
							<br>
						</div>
						<div>
							<font style="font-size: 24px;">exercise &nbsp;4</font>
						</div>
					</section>
					<section class="" data-id="05bd911089c3b0fb4c784ccd28171c9e">
						<h2>MAKING THE ANNOTATION USEFUL</h2>
						<div>
							<br>
						</div>
						<div>create a &nbsp;public genome database to</div>
						<div>present all your work and allow expert curation</div>
						<div>
							<br>
						</div>
						<div>existing GMOD tools make this (fairly) easy:</div>
						<div>
							<br>
						</div>
						<div>
							<ul>
								<li>WebApollo (community annotation editor)</li>
								<li>Tripal (genome data browser, BLAST server)</li>
								<li>G/JBrowse (genome track browser)</li>
								<li>biomart (data mining system)</li>
								<li>PathwayTools (metabolic pathway server)</li>
							</ul>
						</div>
					</section>
					<section class="" data-id="75b7b6a5b87b8d7b6cb08936e016b710">
						<h2>THANKS!</h2>
						<div>
							<br>
						</div>
						<div>Don't forget the resources:</div>
						<div>
							<br>
						</div>
						<div>
							<span style="font-size: 36.185611724853516px; font-style: normal; font-variant: normal;">
								<a href="https://github.com/Blahah/TGAC-2014-genome-annotation">https://github.com/Blahah/TGAC-2014-genome-annotation</a>
							</span>
							<br>
						</div>
					</section>
				</div>


			</div>

			<script src="lib/js/head.min.js"></script>
			<script src="js/reveal.min.js"></script>

			<script>

			// Full list of configuration options available here:
			// https://github.com/hakimel/reveal.js#configuration
			Reveal.initialize({
				controls: true,
				progress: true,
				history: true,
				center: true,

				theme: Reveal.getQueryHash().theme, // available themes are in /css/theme
				transition: Reveal.getQueryHash().transition || 'default', // default/cube/page/concave/zoom/linear/fade/none

				// Parallax scrolling
				// parallaxBackgroundImage: 'https://s3.amazonaws.com/hakim-static/reveal-js/reveal-parallax-1.jpg',
				// parallaxBackgroundSize: '2100px 900px',

				// Optional libraries used to extend on reveal.js
				dependencies: [
				{ src: 'lib/js/classList.js', condition: function() { return !document.body.classList; } },
				{ src: 'plugin/markdown/marked.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/markdown/markdown.js', condition: function() { return !!document.querySelector( '[data-markdown]' ); } },
				{ src: 'plugin/highlight/highlight.js', async: true, callback: function() { hljs.initHighlightingOnLoad(); } },
				{ src: 'plugin/zoom-js/zoom.js', async: true, condition: function() { return !!document.body.classList; } },
				{ src: 'plugin/notes/notes.js', async: true, condition: function() { return !!document.body.classList; } }
				]
			});

</script>

</body>
</html>
